packages/dataviz/src/components/GeoChart/GeoChart.component.tsx

```diff
@@ -0,0 +1,286 @@
+import classNames from 'classnames';
+import {
+	rgb, RGBColor,
+	select as d3select, Selection,
+	scaleLinear, ScaleLinear,
+	geoIdentity, geoPath, GeoPath,
+	zoom as d3zoom,
+} from 'd3';
+import { FeatureCollection } from 'geojson';
+// eslint-disable-next-line import/no-unresolved
+import { Topology } from 'topojson-specification';
+import { feature } from 'topojson-client';
+import throttle from 'lodash/throttle';
+import { Icon } from '@talend/react-components';
+
+import TooltipContent, { TooltipEntry } from '../TooltipContent/TooltipContent.component';
+import styles from './GeoChart.scss';
+
+// Rename ugly d3 types
+type ColorScale = ScaleLinear<RGBColor, RGBColor>;
+type SvgPath = Selection<SVGPathElement, unknown, null, undefined>;
+type Svg = Selection<SVGSVGElement, unknown, null, undefined>;
+type SvgG = Selection<SVGGElement, unknown, null, undefined>;
+
+export interface Entry {
+	key: string;
+	value: number;
+}
+
+export interface GeoChartConfig {
+	topology: Topology;
+	/** the name of the `objects` collection to use (first one will be used if not provided) */
+	layer?: string;
+	labelProperty?: string;
+	valueProperties: string[];
+}
+
+export interface GeoChartProps {
+	chartConfig: GeoChartConfig;
+	data: Entry[];
+	columnName: string;
+	onSelection: (key: string) => void;
+}
+
+interface TooltipPosition {
+	x: number;
+	y: number;
+}
+
+// Don't worry, chart will fit it's container
+const width = 250;
+const height = 200;
+
+// Reuse collator for better performance
+const COLLATOR = new Intl.Collator(undefined, {
+	usage: 'search',
+	sensitivity: 'base',
+	ignorePunctuation: true,
+});
+
+function findEntry(
+	data: Entry[],
+	properties: { [name: string]: string | number },
+	chartConfig: GeoChartConfig,
+): Entry | undefined {
+	// Match configured fields
+	return data.find(entry =>
+		chartConfig.valueProperties.some(
+			property => COLLATOR.compare(entry.key, `${properties[property]}`) === 0,
+		),
+	);
+}
+
+function renderFeature(
+	path: SvgPath,
+	colorScale: ColorScale,
+	setTooltipEntry: (tooltipEntry: TooltipEntry | null) => void,
+	setTooltipPosition: (tooltipPosition: TooltipPosition | null) => void,
+	onSelection: (key: string) => void,
+	entry?: Entry,
+	label?: string,
+): void {
+	path.attr(
+		'class',
+		classNames(styles['geo-chart__feature'], {
+			[styles['geo-chart__feature--disabled']]: !entry,
+		}),
+	);
+	if (entry) {
+		path
+			.attr('data-key', entry.key)
+			.attr('data-value', entry.value)
+			.style('fill', colorScale(+entry.value)?.toString() || '')
+			.on('click', () => onSelection(entry.key))
+			.on('mouseover', d3Event => {
+				// if we have both label and entry key: "label (key): value", otherwise "label: value" or "key: value"
+				const tooltipLabel = `${label || entry.key}${
+					label && entry.key !== label ? ` (${entry.key})` : ''
+				}`;
+				setTooltipEntry({
+					key: tooltipLabel,
+					value: `${entry.value}`,
+				});
+				setTooltipPosition({
```

Comment:
I guess you could use the same state object for this

Reply:
Yes, of course, I find more readable that way thanks to explicit setters but it is fine to combine them into one